### ISSUE TITLE
va-sidenav - Remove header element from component

### ISSUE THREAD

**Issue by amyleadem** [department-of-veterans-affairs/vets-design-system-documentation#4558]
### Description
**The side navigation should not use a header element.** Currently, the side navigation has an optional header prop that generates an `<h2>` element inside the component. After more deeply considering the location of the side navigation on the whole page in both focus order and markup, we realized that adding a header element to the side nav _could_ cause an automated testing failure related to header order. Additionally, because the component lives inside a labeled `<nav>` region, there is enough information for screen reader users to effectively navigate, and adding the component to the header hierarchy did not seem necessary. 

### Must
Recommend switching the `<h2>` element in the component to a `<span>`. Note that this is just a markup change and the element can still keep the current text styling.

### Acceptance Criteria
[Describe what a passing accessibility test would look like, either the code expectation, or the screen reader behavior]

## Type of issue
- [ ] Code review
- [ ] Readability
- [ ] Automated scans
- [ ] Use of color
- [ ] Text resizing, zoom, and magnification
- [x] Screen readers
- [ ] Input and interaction methods
